docs/lang/articles/reference.md

```diff
@@ -241,12 +241,50 @@ The extended form, `assert expression1, expression2`, raises `TaichiAssertionErr
 with `expression2` as the error message.
 
 ### The `pass` statement
+```
+pass_stmt ::=  "pass"
+```
+
+`pass` is a null operation — when it is executed, nothing happens.
+It is useful as a placeholder when a statement is required syntactically, but no code needs to be executed.
 
 ### The `return` statement
+```
+return_stmt ::=  "return" [expression_list]
+```
+
+The return statement may only occur once in a Taichi kernel or a Taichi function,
+and it must be at the bottom of the function body.
+
+If a Taichi kernel or Taichi function has a return type hint,
+it must have a return statement that returns a value other than `None`.
+
+If a Taichi kernel has a return statement that returns a value other than `None`, it must have a return type hint.
+The return type hint for Taichi function is optional but recommended.
+
+A kernel can have at most one return value, which can be a scalar, ti.Matrix, or ti.Vector,
+and the number of elements in the return value does not exceed 30.
```

Comment:
> does not exceed 30.

Explain that this is an implementation detail. It can be relaxed  in the future.